ts/features/newProfile/saga/profile.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * A saga that manages the Profile.
+ */
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import { delay, call, put, select, takeLatest } from "typed-redux-saga/macro";
+import Config from "react-native-config";
+import { getType } from "typesafe-actions";
+import { Millisecond } from "@pagopa/ts-commons/lib/units";
+import { InitializedProfile } from "../../../../definitions/backend/InitializedProfile";
+import { BackendClient } from "../../../api/backend";
+import I18n from "../../../i18n";
+import { sessionExpired } from "../../../store/actions/authentication";
+import {
+  profileLoadFailure,
+  profileLoadSuccess
+} from "../../../store/actions/profile";
+import { ReduxSagaEffect, SagaCallReturnType } from "../../../types/utils";
+import { convertUnknownToError } from "../../../utils/errors";
+import { readablePrivacyReport } from "../../../utils/reporters";
+import { sessionTokenSelector } from "../../../store/reducers/authentication";
+import { authenticationSaga } from "../../../sagas/startup/authenticationSaga";
+import { refreshUserProfileDataRequest } from "../store/actions/profile";
+
+const BACKEND_PROFILE_RETRY_DELAY_INTERVAL = (6 * 1000) as Millisecond;
+const apiUrlPrefix: string = Config.API_URL_PREFIX;
+
+// This function listens for Profile refresh requests and calls the needed saga.
+export function* refreshProfileOnMountOfTheNewProfilePageSaga(): Iterator<ReduxSagaEffect> {
+  yield* takeLatest(getType(refreshUserProfileDataRequest), refreshProfile);
+}
+
+export function* refreshProfile(): Generator<ReduxSagaEffect, void, any> {
+  // Whether the user is currently logged in.
+  const previousSessionToken: ReturnType<typeof sessionTokenSelector> =
+    yield* select(sessionTokenSelector);
+
+  const sessionToken: SagaCallReturnType<typeof authenticationSaga> =
+    previousSessionToken
+      ? previousSessionToken
+      : yield* call(authenticationSaga);
+
+  const backendClient: ReturnType<typeof BackendClient> = BackendClient(
+    apiUrlPrefix,
+    sessionToken
+  );
+
+  // Load the profile info
+  const maybeUserProfile: SagaCallReturnType<typeof loadProfile> = yield* call(
+    loadProfile,
+    backendClient.getProfile
+  );
+
+  if (O.isNone(maybeUserProfile)) {
+    /**
+     * If we didn't succeed,
+     * we try again after waiting some time.
+     */
+    yield* delay(BACKEND_PROFILE_RETRY_DELAY_INTERVAL);
+    yield* put(refreshUserProfileDataRequest());
+  }
```

Comment:
I guess that we should properly handle this scenario avoiding an automatic reload, for example a retry logic in the UI component what do you think?

Reply:
Yup, in this case I'll change a bit how the `ProfileMainScreen` behaves during the REST API call.